src/ActionList/Item.tsx

```diff
@@ -165,49 +169,55 @@ export const Item = React.forwardRef<HTMLLIElement, ActionListItemProps>(
       // hide divider after dividers & group header, with higher importance!
       '[data-component="ActionList.Divider"] + &': {'--divider-color': 'transparent !important'},
       // hide border on current and previous item
-      '&:hover:not([aria-disabled]), &:focus:not([aria-disabled]), &[data-focus-visible-added]:not([aria-disabled])': {
-        '--divider-color': 'transparent',
-      },
-      '&:hover:not([aria-disabled]) + &, &[data-focus-visible-added] + li': {
+      '&:hover:not([aria-disabled]):not([data-inactive]), &:focus:not([aria-disabled]):not([data-inactive]), &[data-focus-visible-added]:not([aria-disabled]):not([data-inactive])':
+        {
+          '--divider-color': 'transparent',
+        },
+      '&:hover:not([aria-disabled]):not([data-inactive]) + &, &[data-focus-visible-added] + li': {
         '--divider-color': 'transparent',
       },
       ...(active ? activeStyles : {}),
     }
 
     const clickHandler = React.useCallback(
       (event: React.MouseEvent<HTMLLIElement>) => {
-        if (disabled) return
+        if (disabled || inactive) return
         onSelect(event, afterSelect)
       },
-      [onSelect, disabled, afterSelect],
+      [onSelect, disabled, inactive, afterSelect],
     )
 
     const keyPressHandler = React.useCallback(
       (event: React.KeyboardEvent<HTMLLIElement>) => {
-        if (disabled) return
+        if (disabled || inactive) return
```

Comment:
I'm not calling any handlers passed to `onSelect` for inactive items. This would also make it impossible to render a dialog when the user clicks the item. I think this is a reasonable limitation to begin with, and we can always update the implementation to support click/selection handlers so users can show a `Dialog` if we find we need to.

Do you agree that it's safe to prohibit `onSelect` for now?

Reply:
Here's a draft PR to update the interface guidelines: https://github.com/primer/design/pull/654

I'll close it if we think we _do_ need to support click/selection behavior on inactive items.
I'll merge it if we don't think we need to support click/selection behavior on inactive items.